wa/commands/process.py

```diff
@@ -0,0 +1,107 @@
+#    Copyright 2014-2018 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+
+from wa import Command
+from wa import discover_wa_outputs
+from wa.framework import pluginloader
+from wa.framework.exception import CommandError
+from wa.framework.output import RunOutput
+from wa.framework.output_processor import ProcessorManager
+
+class ProcessContext(object):
+
+    def __init__(self):
+        self.run_output = None
+        self.target_info = None
+        self.job_output = None
+
+class ProcessCommand(Command):
+
+    name = 'process'
+    description = 'Process the output from previously run workloads.'
+
+    def __init__(self, subparsers):
+        self.loader = pluginloader
```

Comment:
This is not used and is not necessary; use `config.plugin_cache` instead.

Reply:
I tried that, while correct that the quoted point is not required, using `config.plugin_cache` instead of `pluginloader` when initialising the `ProcessorManager` results in it not being able to perform the `install` operation, happy to try to work around this if you think that is still the correct course of action.

While I did need that config point clarifying regardless, further investigation of the problem I was having revealed it was not the cause. The cause is that in the `Configuration` class in `wa.framework.configuration.core` has a `from_pod` function that loads the line: 

> "boot_parameters": null, 

from jobs.json as an empty dict `{}`, which results in `_update_spec` in the sqlite processor to fail the `json.dumps(spec.boot_parameters)` call because it is not serializable. Is there a reason the pod load works this way? During the run the `boot_parameters` value is `None`

In the meantime I will continue to test what I have plus check whether the config is working, the database is created before the failure point above for the sqlite processor.

Reply:
erm, I'll need to look into this; an empty dict should be serializable.

Reply:
ok, I see the problem; it's actually an `obj_dict`, so needs to be `to_pod()`'d. Will fix.

Reply:
Should be fixed by https://github.com/ARM-software/workload-automation/pull/709/files

Reply:
Thank you, I'll grab that and retest

Reply:
That does seem to have fixed it thanks